### PR TITLE
fix(scripts): remove extract_summary_status() override in recover-state.sh

### LINKED ISSUE #428: recover-state.sh overrides summary-utils parser with fragile subprocess pipeline
## Problem statement

`script/recover-state.sh` sources `scripts/summary-utils.sh`, but then immediately redefines `extract_summary_status()` using the older multi-process `sed | head | tr` pipeline. The main fix for issue #414 hardened the hot path by moving summary status extraction to a pure-bash helper specifically to avoid intermittent failures under heavily parallel verification runs. Because `recover-state.sh` overrides that helper, it still carries the same fragility class in a downstream recovery path.

Concrete symptom:
- `scripts/summary-utils.sh` now provides a pure-bash `extract_summary_status()` implementation.
- `scripts/recover-state.sh` lines 13-21 redefine `extract_summary_status()` locally with the old subprocess-heavy parser.
- Under transient process/fd pressure, `recover-state.sh` can still misread summary status or reconstruct stale plan state even though the primary `phase-detect` path was hardened.

## Root cause analysis

The recovery script duplicated parser logic instead of reusing the shared helper it already sources. That creates a split-brain implementation:
- shared helper: pure-bash, brownfield-tolerant parser in `scripts/summary-utils.sh`
- recovery-only override: older subprocess pipeline in `scripts/recover-state.sh`

This breaks the intended invariant that all runtime consumers use the same summary status extraction semantics and failure characteristics.

## Affected components

- `scripts/recover-state.sh`
- `scripts/summary-utils.sh`
- any recovery flows that depend on `extract_summary_status()` while reconstructing phase state
- relevant contract coverage for recovery behavior

## Acceptance criteria

1. `scripts/recover-state.sh` does not override `extract_summary_status()` with its own subprocess-based implementation.
2. Recovery state reconstruction uses the shared summary parser from `scripts/summary-utils.sh`.
3. Recovery behavior remains correct for at least these inputs: normal frontmatter, CRLF line endings, quoted status values, whitespace-padded status values, and leading blank line / BOM before frontmatter.
4. Existing recovery-related tests continue to pass.
5. If recovery-specific tests do not currently lock this in, add or update targeted tests so a future parser split is caught automatically.

## Scope boundary

This issue is about the downstream `recover-state.sh` parser override only. It is not asking to rework unrelated phase-detect, qa-result-gate, or sharding logic unless a minimal shared-helper reuse change requires small adjacent updates.

### PR DESCRIPTION
**Pull request by dpearson2699**
## Linked Issue

Fixes #428

Also filed #431 as a follow-up observation (session-start.sh reconcile block has the same old parser — out of scope for this PR).

## What

Removes the `extract_summary_status()` override in `scripts/recover-state.sh` that duplicated parser logic after sourcing `scripts/summary-utils.sh`. Adds edge-case BATS tests and a contract check to prevent future overrides.

**Files modified:**
- `scripts/recover-state.sh` — Removed the subprocess-based `sed | head | tr` override of `extract_summary_status()`; recovery now uses the shared pure-bash parser from `summary-utils.sh`
- `tests/recover-state-integration.bats` — Added 4 edge-case tests (CRLF, quoted status, whitespace-padded, BOM+leading blank line)
- `testing/verify-summary-utils-contract.sh` — Added contract check that prevents consumer scripts from overriding `extract_summary_status()` after sourcing the shared helper

## Why

PR #429 (fixing #414) hardened the hot path by moving summary status extraction to a pure-bash helper in `summary-utils.sh` to avoid intermittent failures under parallel verification. However, `recover-state.sh` was still overriding that helper with the older fragile `sed | head | tr` pipeline, creating a split-brain implementation. The correct architectural fix is to remove the override entirely and let the shared helper handle all edge cases uniformly.

## How

- **`scripts/recover-state.sh`**: Removed the 5-line `extract_summary_status()` function override (lines 18-22 in the old code). The `is_plan_finalized()` bridge wrapper to `is_summary_terminal()` is retained. The else-block fallback stub (for when `summary-utils.sh` is missing) is unchanged.
- **`tests/recover-state-integration.bats`**: Added 4 tests that exercise `recover-state.sh` with CRLF line endings, quoted status values, whitespace-padded status values, and BOM+leading blank line before frontmatter. These validate criterion 3.
- **`testing/verify-summary-utils-contract.sh`**: Added a loop that counts `extract_summary_status()` definitions in each consumer script; fails if a script that sources `summary-utils.sh` also defines the function more than once (the else-block fallback counts as 1; an override in the sourced branch would be 2+). QA round 1 also fixed the `grep -cE ... || echo "0"` pattern to use proper exit-code handling.

## Acceptance Criteria Verification

1. **`scripts/recover-state.sh` does not override `extract_summary_status()`** — Satisfied: the override is removed at line 18, replaced with a comment. Verified by contract check.
2. **Recovery uses shared parser from `summary-utils.sh`** — Satisfied: line 15 sources `summary-utils.sh`, line 89 calls `extract_summary_status()` from the shared lib.
3. **Correct for all input classes** — Satisfied: 4 new edge-case tests (CRLF, quoted, whitespace-padded, BOM) + existing normal frontmatter test all pass.
4. **Existing recovery tests pass** — Satisfied: 30/30 recover-state-integration.bats tests pass, 2853/2853 total BATS.
5. **Contract check prevents future split** — Satisfied: new check in `verify-summary-utils-contract.sh` catches overrides. 48/48 contract assertions pass.

## Testing

- [x] `bash testing/run-all.sh` — 39/39 contract checks, 2853/2853 BATS, lint clean
- [x] `bats tests/recover-state-integration.bats` — 30/30 (including 4 new edge-case tests)
- [x] `bash testing/verify-summary-utils-contract.sh` — 48/48 (including new override check)

## QA Summary

- **Primary QA (Claude)**: 3 rounds. Round 1: 1 low finding (grep fallback pattern in contract check — fixed in `dd8767e`). Rounds 2-3: zero findings.
- **Cross-model QA (GPT-5.4)**: 1 round. Zero contract/regression findings. 1 low observation (session-start.sh reconcile block — filed as #431, out of scope).
- **Copilot PR review**: Pending (Phase 4.5)
- **Findings fixed**: 1 (QA round 1). **False positives**: 0.

### CODE REVIEW

File: testing/verify-summary-utils-contract.sh [hunk 1]
```diff
@@ -476,6 +476,24 @@ for script in phase-detect.sh state-updater.sh recover-state.sh qa-gate.sh file-
...
+# definitions that coexist with the sourced helper create split-brain parsing.
+for script in phase-detect.sh state-updater.sh recover-state.sh qa-gate.sh file-guard.sh; do
+  # Count function definitions of extract_summary_status()
+  def_count=$(grep -cE '^\s*extract_summary_status\s*\(\)' "$ROOT/scripts/$script" 2>/dev/null) || def_count=0
+  # Count source lines for summary-utils.sh
+  source_count=$(grep -cE '^\s*\.\s+.*summary-utils\.sh' "$ROOT/scripts/$script" 2>/dev/null) || source_count=0
+  # If the script sources summary-utils.sh AND defines extract_summary_status() more than once
+  # (the fallback else-stub), that means there's an override in the sourced branch.
+  # If it doesn't source summary-utils.sh, any definition is a standalone (not an override).
+  if [ "$source_count" -gt 0 ] && [ "$def_count" -gt 1 ]; then
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply testing/verify-summary-utils-contract.sh:490]: Good catch. Fixed in c37486c — updated the `source_count` regex from `'^\s*\.\s+.*summary-utils\.sh'` to `'^\s*(\.|source)\s+.*summary-utils\.sh'` so it matches both `. file` and `source file` forms.

Note: the earlier `if grep -q 'summary-utils\.sh'` check (line 463) already uses a broad pattern that catches all sourcing forms. The override detection's `source_count` was the only place that was narrower than intended.

File: tests/recover-state-integration.bats [hunk 2]
```diff
@@ -640,3 +640,61 @@ STATE
...
+}
+
+@test "recover-state: parses SUMMARY with leading blank line and BOM" {
+  cd "$TEST_TEMP_DIR"
+  local tmp
+  tmp=$(mktemp)
+  jq '.event_recovery = true' .vbw-planning/config.json > "$tmp" && mv "$tmp" .vbw-planning/config.json
+
+  echo "title: Build UI" > .vbw-planning/phases/01-setup/01-01-PLAN.md
+  printf '\xef\xbb\xbf\n---\nstatus: complete\n---\n# Summary\n' > .vbw-planning/phases/01-setup/01-01-SUMMARY.md
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/recover-state-integration.bats:695]: False positive — `printf '\xef\xbb\xbf'` in both bash and zsh correctly emits the raw bytes EF BB BF. Verified locally:

```
$ printf '\xef\xbb\xbf' | xxd | head -1
00000000: efbb bf                                  ...
```

The `\x` escape sequences in `printf` are interpreted as hex byte values in all POSIX-conforming shells. The test is correct as written and already exercises BOM handling.

File: testing/verify-summary-utils-contract.sh [hunk 3]
```diff
@@ -476,6 +476,24 @@ for script in phase-detect.sh state-updater.sh recover-state.sh qa-gate.sh file-
...
 done
 
+# Verify no consumer script overrides extract_summary_status() after sourcing summary-utils.sh.
+# Fallback stubs in else-blocks (for when summary-utils.sh is missing) are allowed — only
+# definitions that coexist with the sourced helper create split-brain parsing.
+for script in phase-detect.sh state-updater.sh recover-state.sh qa-gate.sh file-guard.sh; do
+  # Count function definitions of extract_summary_status()
+  def_count=$(grep -cE '^\s*extract_summary_status\s*\(\)' "$ROOT/scripts/$script" 2>/dev/null) || def_count=0
+  # Count source lines for summary-utils.sh (matches both `. file` and `source file`)
+  source_count=$(grep -cE '^\s*(\.|source)\s+.*summary-utils\.sh' "$ROOT/scripts/$script" 2>/dev/null) || source_count=0
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply testing/verify-summary-utils-contract.sh]: Good catch — `\s` is a GNU extension, not POSIX ERE. Fixed in 7545dfd by replacing `\s` with `[[:space:]]` in both the `def_count` and `source_count` grep patterns. This ensures the contract check works correctly on both BSD grep (macOS) and GNU grep (Linux CI).
